proof/courier.go

```diff
@@ -849,6 +921,206 @@ func (h *HashMailCourier) SetSubscribers(
 // proof.Courier interface.
 var _ Courier = (*HashMailCourier)(nil)
 
+// UniverseRpcCourier is a universe RPC proof courier service handle. It
+// implements the Courier interface.
+type UniverseRpcCourier struct {
+	// recipient describes the recipient of the proof.
+	recipient Recipient
+
+	// client is the RPC client that the courier will use to interact with
+	// the universe RPC server.
+	client unirpc.UniverseClient
+
+	// deliveryLog is the log that the courier will use to record the
+	// attempted delivery of proofs to the receiver.
+	deliveryLog DeliveryLog
```

Comment:
Would assume this now plays a pretty minimal role. Only thing can think of is if the server is down for w/e reason (unreachable).

Reply:
By mistake I actually didn't use this field, so I'm going to modify this PR to make use of it now.